src/devices/colordistancesensor.ts

```diff
@@ -2,90 +2,111 @@ import { Device } from "./device";
 
 export class ColorDistanceSensor extends Device {
 
+    private static modes: { [name: string]: IDeviceMode } = {
+        COLOR: {
+            /**
+             * Emits when a color sensor is activated.
+             * @event ColorDistanceSensor#color
+             * @param {string} port
+             * @param {Color} color
+             */
+            input: true,
+            event: "color",
+            values: { type: ValueType.UInt8, count: 1, min: 0, max: 255 },
+            num: {
+                [HubType.BOOST_MOVE_HUB]: 0x00,
+                [HubType.CONTROL_PLUS_HUB]: 0x00,
+                [HubType.POWERED_UP_HUB]: 0x00,
+                [HubType.WEDO2_SMART_HUB]: 0x00
+            }
+        },
+        PROX: {
+            input: true,
+            event: "distance",
+            values: { type: ValueType.UInt8, count: 1, min: 0, max: 10 },
+            num: {
+                [HubType.BOOST_MOVE_HUB]: 0x01,
+                [HubType.CONTROL_PLUS_HUB]: 0x01,
+                [HubType.POWERED_UP_HUB]: 0x01
+            }
+        },
+        COUNT: {
+            input: true,
+            event: "count",
+            values: { type: ValueType.UInt8, count: 1, min: 0 },
+            num: {
+                [HubType.BOOST_MOVE_HUB]: 0x02,
+                [HubType.CONTROL_PLUS_HUB]: 0x02,
+                [HubType.POWERED_UP_HUB]: 0x02
+            }
+        },
+        REFLT: {
+            input: true,
+            event: "reflectivity",
+            values: { type: ValueType.UInt8, count: 1, min: 0, max: 100 },
+            num: {
+                [HubType.BOOST_MOVE_HUB]: 0x03,
+                [HubType.CONTROL_PLUS_HUB]: 0x03,
+                [HubType.POWERED_UP_HUB]: 0x03
+            }
+        },
+        AMBI: {
+            input: true,
+            event: "luminosity",
+            values: { type: ValueType.UInt8, count: 1, min: 0, max: 100 },
+            num: {
+                [HubType.BOOST_MOVE_HUB]: 0x04,
+                [HubType.CONTROL_PLUS_HUB]: 0x04,
+                [HubType.POWERED_UP_HUB]: 0x04
+            }
+        },
+        SOME_OUTPUT_MODE: {
+            input: false,
+            num: {}
+        },
+        RGB_I: {
+            input: true,
+            event: "rgb",
+            values: { type: ValueType.UInt8, count: 3, min: 0, max: 255 },
+            num: {
+                [HubType.BOOST_MOVE_HUB]: 0x06,
+                [HubType.CONTROL_PLUS_HUB]: 0x06,
+                [HubType.POWERED_UP_HUB]: 0x06
+            }
+        },
+        OTHER_OUTPUT_MODE: {
+            input: false,
+            num: {}
+        },
+        SPEC_1: {
+            input: true,
+            event: "colorAndDistance",
+            values: { type: ValueType.UInt8, count: 4, min: 0, max: 255 },
+            num: {
+                [HubType.BOOST_MOVE_HUB]: 0x08,
+                [HubType.CONTROL_PLUS_HUB]: 0x08,
+                [HubType.POWERED_UP_HUB]: 0x08
+            }
+        }
+    };
```

Comment:
We could add here some transformation function that get data array in parameters and return transformed array. It could do some calculation for distance or simply remove some values (color and distance).